packages/core/src/js/integrations/default.ts

```diff
@@ -136,10 +135,13 @@ export function getDefaultIntegrations(options: ReactNativeClientOptions): Integ
     (options._experiments && typeof options._experiments.replaysOnErrorSampleRate === 'number') ||
     (options._experiments && typeof options._experiments.replaysSessionSampleRate === 'number')
   ) {
-    integrations.push(notWeb() ? mobileReplayIntegration() : browserReplayIntegration());
-    if (!notWeb()) {
+    if (isWeb()) {
+      // We can't create and add browserReplayIntegration as it overrides the users supplied one
+      // The browser replay integration works differently than the rest of default integrations
       (options as BrowserOptions).replaysOnErrorSampleRate = options._experiments.replaysOnErrorSampleRate;
       (options as BrowserOptions).replaysSessionSampleRate = options._experiments.replaysSessionSampleRate;
+    } else {
+      integrations.push(mobileReplayIntegration());
```

Comment:
I think that's unpredictable behavior we should avoid, if they added it we should respect it.

Sidenote: The integration check if it can run, it not it will gracefully disables itself.